compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -782,34 +782,53 @@ object RefChecks {
         report.error(problem(), clazz.srcPos)
       }
 
-      // check that basetype and subtype agree on types of trait parameters
+      // Checks base types of the current clazz against each other
       //
-      // I.e. trait and class parameters not only need to conform to the expected
-      // type of the corresponding base-trait, but also to the type as seen by the
-      // inheriting subtype.
-      def checkTraitParametersOK() = for {
-        parent <- clazz.info.parents
-        parentSym = parent.classSymbol
-        if parentSym.isClass
-        cls = parentSym.asClass
-        if cls.paramAccessors.nonEmpty
-        param <- cls.paramAccessors
-      } {
-        val tpeFromParent = parent.memberInfo(param)
-        val tpeFromClazz = clazz.thisType.memberInfo(param)
-        if (!(tpeFromParent <:< tpeFromClazz)) {
-          val msg =
-            em"""illegal parameter: The types of $param do not match.
-                |
-                |  $param in $cls has type: $tpeFromParent
-                |  but $clazz expects $param to have type: $tpeFromClazz"""
-
-          report.error(msg, clazz.srcPos)
+      // In particular, it checks that there are no two base classes with
+      // different type instantiations.
+      //
+      // ported from Scala 2:
+      //   https://github.com/scala/scala/blob/9bb659e62a9239c01aec14c171f8598bb1a576fe/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala#L834-L883
+      def validateBaseTypes(): Unit = {
+        val tpe = clazz.thisType // in Scala 2 this was clazz.tpe
+        val seenParents = mutable.HashSet[Type]()
+        val baseClasses: List[ClassSymbol] = clazz.info.baseClasses
+
+        // tracks types that we have seen for a particular base class in baseClasses
+        val seenTypes = mutable.Map.empty[Symbol, List[Type]]
+
+        // validate all base types of a class in reverse linear order.
+        def register(tp: Type): Unit = {
+          val baseClass = tp.typeSymbol
+          if (baseClasses contains baseClass) {
```

Comment:
Scala 2 has an additional check, that honestly, I don't quite understand: 

https://github.com/scala/scala/blob/9bb659e62a9239c01aec14c171f8598bb1a576fe/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala#L853-L854

Reply:
It was introduced here: https://github.com/scala/scala/commit/c0450f0c12f265674bc657cfb469778cd35d1c40

Reply:
Looks like it's a performance thing so we can safely ignore it.